### PR TITLE
Feat: Type definitions for @reach/router hooks

### PR DESCRIPTION
**Pull request by HenryKenya**
**Links to documentation**
- [useLocation](https://reach.tech/router/api/useLocation)
- [useMatch](https://reach.tech/router/api/useMatch)
- [useParams](https://reach.tech/router/api/useParams)
- [useNavigate](https://reach.tech/router/api/useNavigate)

**Contribution**
[react-router_v1.1.x]: add type definition for hooks (useParams, useNavigate, useMatch), remove `location` key in `UseLocation` type

### CODE REVIEW

**Comment by HenryKenya** [comment]
@GAntoine  added type definitions for other hooks.

File: definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/router_v1.1.x.js [hunk 1]
```diff
@@ -138,5 +140,12 @@ declare module '@reach/router' {
...
   declare export function redirectTo(uri: string): void;
 
-  declare export function useLocation(): $PropertyType<UseLocation, 'location'>;
+  declare export function useLocation(): UseLocation;
+
+  declare export function useMatch(path: string): UseMatch;
+
+  declare export function useNavigate(): NavigateFn;
+
+  declare export function useParams(): { [key: string]: any };
```
Comment (pascalduez) [inline definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/router_v1.1.x.js:149]:
```suggestion
  declare export function useParams(): { [key: string]: mixed, ... };
```

File: definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/test_router_v1.1.x.js [hunk 2]
```diff
@@ -269,9 +273,43 @@ describe('@reach/router', () => {
...
   });
 
-  describe('react hook', () => {
+  describe('react hooks', () => {
     it('useLocation', () => {
       const location: UseLocation = useLocation();
     });
+
+    it('useParams', () => {
+      const params: { [key: string]: any } = useParams();
```
Comment (pascalduez) [inline definitions/npm/@reach/router_v1.1.x/flow_v0.63.x-v0.103.x/test_router_v1.1.x.js:282]:
```suggestion
      const params: { [key: string]: mixed, ... } = useParams();
```

**Comment by HenryKenya** [comment]
@pascalduez  , update type definition for `useParams`

**Review by pascalduez — APPROVED** [review]

**Comment by HenryKenya** [comment]
@GAntoine is it possible to merge the PR?

**Pull request merged**